Add possibility to hook into the config initialization. (#2590)
domaindrivendev/Swashbuckle.AspNetCore: README.md

});
```

### Change Swagger UI Config ###

> Swagger UI has a big variety of options and a mighty plugin API allowing customizations. 

Reviewer: ```suggestion
Swagger UI has a large variety of options and a mighty plugin API allowing customizations. 
```

To customize the Swagger UI (and OAuth) configuration objects before they are passed into the Swagger UI initialization,
inject a custom JavaScript file which defines a global function with the following signature.

> In many cases this feature allows to avoid using a custom `index.html` in your project.

Reviewer: ```suggestion
In many cases this feature allows you to avoid using a custom `index.html` in your project.
```

```js
/**
 * @param configObject See https://swagger.io/docs/open-source-tools/swagger-ui/usage/configuration/
 * @param oauthConfigObject See https://swagger.io/docs/open-source-tools/swagger-ui/usage/oauth2/
 */
function initConfig(configObject, oauthConfigObject) { }
```

> Through this mechanism any options that Swagger UI supports can be set and also plugins can be registered. 

Reviewer: ```suggestion
Through this mechanism any options that Swagger UI supports can be set, and also plugins can be registered. 
```

Swagger UI is written with React so it can be a bit tricky to write custom components. 

> Here a full example which adds a custom section to the parameters of all operations:

Reviewer: ```suggestion
Here is a full example which adds a custom section to the parameters of all operations:
```

```js
// swagger-ui.js
// https://swagger.io/docs/open-source-tools/swagger-ui/customization/plugin-api/
function configInit(config) {

>     if(!config.plugins) {

Reviewer: ```suggestion
    if (!config.plugins) {
```

        config.plugins = [];
    }
    config.plugins.push(TopbarPlugin);
}

const TopbarPlugin = (system) => {
    return {

>         // Register an own React component 

Reviewer: ```suggestion
        // Register a custom React component 
```

        components: {
            CustomOperationDetails: class CustomOperationDetails extends system.React.Component {
                render() {
                    const hello = this.props?.operation?.get('x-hello');

>                     if(!hello) {

Reviewer: ```suggestion
                    if (!hello) {
```

                        return '';
                    }
                    return system.React.createElement('div',
                        { className: 'hello' },
                        hello
                    );
                }
            }
        },
        wrapComponents: {

>             // wrap the parameters component
>             parameters: Original => (props => {
>                 // get the custom React component

Reviewer: ```suggestion
            // Wrap the parameters component
            parameters: Original => (props => {
                // Get the custom React component
```

                const CustomOperationDetails = props.getComponent('CustomOperationDetails');

>                 // Use the non JSX React API (https://reactjs.org/docs/react-without-jsx.html)

Reviewer: ```suggestion
                // Use the non JSX React API (https://react.dev/reference/react/createElement#creating-an-element-without-jsx)
```

                // to create a wrapper div holding our own component and the original parameters component
                return system.React.createElement('div',
                    { className: "parameters-wrap"},
                    system.React.createElement(CustomOperationDetails, props),
                    system.React.createElement(Original, props)

>                 );

Reviewer: This example looks out-of-date compared to the example in the non-obsolete React documentation.

            })
        }
    }  
};
```

```csharp

> app.UseSwaggerUI(options = > 
> {
>     options.InjectJavascript("swagger-ui.js");
> });
> ```
> 
> 

Reviewer: ```suggestion
app.UseSwaggerUI(options => options.InjectJavascript("swagger-ui.js"));
```

```
